File: CSharpDefinitions/Samples/dxChartSample.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpDefinitions.Samples.dxChartSample {

    public interface dxChart {

        CommonSeriesSettings commonSeriesSettings { get; }
        Union<BarSeries, LineSeries, Union<BarSeries, LineSeries>[]> series { get; }
        dxChart instance();
    }

    [InjectedType]
    public interface CommonSeriesSettings {

        [DefaultValue("line")]
        string type { get; }
        LineSeriesSettings line { get; }
        BarSeriesSettings bar { get; }

        SeriesSettings.valueField valueField { get; set; }
        SeriesSettings.cornerRadius cornerRadius { get; set; }
        SeriesSettings.width width { get; set; }
    }


    public interface LineSeries : LineSeriesSettings {

        [DefaultValue("line")]
        string type { get; }
        string name { get; }
    }

    public interface BarSeries : BarSeriesSettings {

        [DefaultValue("bar")]
        string type { get; }
        string name { get; }
    }

    public interface LineSeriesSettings {

        SeriesSettings.valueField valueField { get; set; }
        SeriesSettings.width width { get; set; }
    }

    public interface BarSeriesSettings {

        SeriesSettings.cornerRadius cornerRadius { get; }
        SeriesSettings.valueField valueField { get; }
    }

    namespace SeriesSettings {

        [DefaultValue("val")]
        public class valueField : Alias<string> { }

        [DefaultValue(2)]
        public class width : Alias<int> { }

        [DefaultValue(0)]
        public class cornerRadius : Alias<int> { }

    }
}

File: JSDoc/Samples/dxChartSample.js
/**
 * @class
 */
var dxChart = {

    /**
     * @memberof dxChart
     * @mixes seriesSettings
     * @type {object}
     */
    commonSeriesSettings: {

        /**
         * @memberof dxChart.commonSeriesSettings
         * @type {string}
         * @default line
         */
        type,

        /**
         * @memberof dxChart.commonSeriesSettings
         * @type {LineSeriesSettings}
         */
        line,

        /**
         * @memberof dxChart.commonSeriesSettings
         * @type {BarSeriesSettings}
         */
        bar
    },

    /**
     * @method
     * @memberof dxChart
     * @type dxChart
     * @returns {dxChart} returns instance
     */
    instance: function () {
        return "";
    },

    /**
     * @memberof dxChart
     * @type BarSeries|LineSeries|Array.<BarSeries|LineSeries>
     */
    series: [{}]
};

/**
 * @class
 * @extends LineSeriesSettings
 */
var LineSeries = {
    /**
     * @memberof LineSeries
     * @type {string}
     */
    name,

    /**
     * @memberof LineSeries
     * @type {string}
     * @default line
     */
    type
};

/**
 * @class
 * @extends BarSeriesSettings
 */
var BarSeries = {
    /**
     * @memberof BarSeries
     * @type {string}
     */
    name,

    /**
     * @memberof BarSeries
     * @type {string}
     * @default bar
     */
    type
};

/**
 * @class LineSeriesSettings
 * @mixes seriesSettings.valueField
 * @mixes seriesSettings.width
 */

/**
 * @class BarSeriesSettings
 * @mixes seriesSettings.valueField
 * @mixes seriesSettings.cornerRadius
 */

/**
 * @mixin
 */
var seriesSettings = {

    /**
     * @memberof seriesSettings
     * @type {string}
     * @default val
     */
    valueField,

    /**
     * @memberof seriesSettings
     * @type {int}
     * @default 2
     */
    width,

    /**
     * @memberof seriesSettings
     * @type {int}
     * @default 0
     */
    cornerRadius
};